src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -55,6 +56,46 @@ auto CMitsubaMaterialCompilerFrontend::getTexture(const CElementTexture* _elemen
     return { nullptr, nullptr, scale };
 }
 
+
+auto CMitsubaMaterialCompilerFrontend::getEmissionProfile(const CElementEmissionProfile* _element) const -> tex_ass_type
+{
+    asset::IAsset::E_TYPE types[2]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+    const auto key = SContext::emissionProfileCacheKey(*_element);
+    auto viewBundle = m_loaderContext->override_->findCachedAsset(key, types, m_loaderContext->inner, 0u);
+    if (!viewBundle.getContents().empty())
+    {
+        auto view = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(viewBundle.getContents().begin()[0]);
+
+        types[0] = asset::IAsset::ET_SAMPLER;
+        const std::string samplerKey = m_loaderContext->samplerCacheKey(m_loaderContext->emissionProfileSamplerParams(*_element));
+        auto samplerBundle = m_loaderContext->override_->findCachedAsset(samplerKey, types, m_loaderContext->inner, 0u);
+        assert(!samplerBundle.getContents().empty());
+        auto sampler = core::smart_refctd_ptr_static_cast<asset::ICPUSampler>(samplerBundle.getContents().begin()[0]);
+
+        return { view, sampler, 1.0f };
+    }
+    return { nullptr, nullptr, 1.0f };
+}
+
+CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend::createEmitterNode(asset::material_compiler::IR* ir, const CElementEmitter* _emitter, core::matrix4SIMD transform) {
+    assert(_emitter->type == CElementEmitter::AREA);
+    auto* res = ir->allocNode<asset::material_compiler::IR::CEmitterNode>();
+    if (_emitter->area.emissionProfile) {
+        asset::material_compiler::IR::CEmitterNode::EmissionProfile profile;
+        asset::material_compiler::IR::INode::STextureSource tex;
+        std::tie(tex.image, tex.sampler, tex.scale) = getEmissionProfile(_emitter->area.emissionProfile);
+        assert(tex.image && "emission profile not loaded");
```

Comment:
thou shalt not assert to handle failures, you will log and provide a default.

Also can you assert `tex.scale==1.f`